ale_linters/c/cppcheck.vim

```diff
@@ -18,10 +28,21 @@ function! ale_linters#c#cppcheck#GetCommand(buffer) abort
     \   ? '--project=compile_commands.json '
     \   : ''
 
+    " if the buffer is modified or we haven't found compile_commands.json,
+    " include the file's directory.
+    if l:modified || empty(l:compile_commmands_path)
+      " Get path to this buffer so we can include it into cppcheck with -I
+      " This could be expanded to get more -I directives from the compile
+      " command in compile_commands.json, if it's found.
+      let l:buffer_path = fnamemodify(bufname(a:buffer), ':p:h')
+      let l:buffer_path_include = ' -I'.l:buffer_path
```

Comment:
```suggestion
      let l:buffer_path_include = ' -I' . ale#Escape(l:buffer_path)
```

Escape the path, in case it contains spaces, etc.

Reply:
I'll escape it. What do you mean by duplicate code? The GetCommand function for C and CPP is very similar, do you mean that? Where would this common function live?